lib/facade_converter.ts

```diff
@@ -85,6 +85,31 @@ export class FacadeConverter extends base.TranspilerBase {
     return handler && !handler(pa);
   }
 
+  visitTypeName(typeName: ts.EntityName) {
+    if (typeName.kind !== ts.SyntaxKind.Identifier) {
+      this.visit(typeName);
+      return;
+    }
+    var ident = base.ident(typeName);
+    if (this.tc) {
+      var symbol = this.tc.getSymbolAtLocation(typeName);
+      if (!symbol) {
```

Comment:
You could be a bit less draconic by checking first if the type name is ever substituted at all, and only report an error if a collision could happen - same as we do for property names.